Added an action button to change your entire chameleon outfit at once

Cruix opened the pull request:
:cl:
add: Added a button to chameleon items to change your entire chameleon outfit to the standard outfit of a selected job.
/:cl:

The extra button appears when you first equip a chameleon item, and disappears when your remove the last chameleon item you are holding. If you pick up additional chameleon items, the button will not be duplicated.

[why]: # 90% of the time, chameleon items are used to emulate the outfit of a particular job, and this is far more convenient than having to scroll through the gigantic, unsorted menu for each individual item to find the names of the correct items.

Cruix committed:
Added an action button to change your entire chameleon outfit at once

PKPenguin321 commented:
Nice

SpaceManiac commented on code/modules/clothing/chameleon.dm:
```diff
@@ -68,6 +68,69 @@
+	button_icon_state = "chameleon_outfit"
+	var/list/outfit_options //By default, this list is shared between all instances.\
+							It is not static because if it were, subtypes would not be able to have their own.\
+							If you ever want to edit it, copy it first.
```
>`\`

Please don't do this to me.

SpaceManiac reviewed (changes requested):
Great feature, should make for a lot of fun. Not OP because the outfit is not necessarily perfect, but will help traitors not forget really easy stuff. Just some nitpicks:

* Please do not backslash in comments
* "Shaft Miner (Lavaland + Equipment)" should include the explorer's hood
* Jobs with stamps should include their stamps
* I think the action button animates a little too rapidly

Cruix committed:
Made chameleon stamps actual chameleon items
Added stamps to bonus chameleon items on outfits that should have them
Made clothing work with hooded suits

Cruix commented:
Had to rewrite stamps so that they are actual chameleon items.
Apparently hooded suits and hardsuits are not related, they just use nearly identical code with slightly different proc and var names. I might fix that later.
The action button icon changing quickly is the whole point.

Cruix committed:
no message

SpaceManiac reviewed (approved):
Personally I find it a little bit visually distracting but that's just a matter of opinion. Looks good to go.
